lib/heex_formatter/phases/format.ex

```diff
@@ -82,23 +83,42 @@ defmodule HeexFormatter.Phases.Format do
         end
       end
 
-    indentation = if self_closed?, do: state.indentation, else: state.indentation + 1
+    state =
+      if self_closed? do
+        state
+      else
+        if tag == "script" do
+          %{state | mode: :script}
+        else
+          %{state | indentation: state.indentation + 1}
+        end
+      end
+
+    %{state | html: state.html <> "\n" <> tag_opened}
+  end
 
-    %{state | html: state.html <> "\n" <> tag_opened, indentation: indentation}
+  defp token_to_string({:text, text, _meta}, %{mode: :script} = state) do
+    indent = indent_expression(state.indentation)
+    %{state | html: state.html <> indent_script(indent, text)}
   end
 
   defp token_to_string({:text, text, _meta}, state) do
-    text =
-      case state.previous_token do
-        {:eex_tag_render, _tag, _meta} ->
-          " " <> String.trim(text)
-
-        _token ->
-          indent = indent_expression(state.indentation)
-          "\n" <> indent <> String.trim(text)
-      end
+    case state.previous_token do
+      {:eex_tag_render, _tag, _meta} ->
+        %{state | html: state.html <> " " <> String.trim(text)}
 
-    %{state | html: state.html <> text}
+      _token ->
+        indent = indent_expression(state.indentation)
+        %{state | html: state.html <> "\n" <> indent <> String.trim(text)}
+    end
+  end
+
+  defp token_to_string({:tag_close, "script" = tag, _meta}, state) do
+    indentation = state.indentation
+    indent = indent_expression(indentation)
+    tag_closed = "\n#{indent}</#{tag}>"
+
+    %{state | html: state.html <> tag_closed, indentation: indentation}
```

Comment:
```suggestion
```

I guess we don't need since we are not going to indent stuff within `script` tag.

Reply:
erased